File: SDKs/googleapis.com/siteVerification/v1/python/src/sdk/models/shared/siteverificationwebresourcegettokenrequest.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class SiteVerificationWebResourceGettokenRequestSite:
    r"""SiteVerificationWebResourceGettokenRequestSite
    The site for which a verification token will be generated.
    """
    
    identifier: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('identifier') }})
    type: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

@dataclass_json
@dataclass
class SiteVerificationWebResourceGettokenRequest:
    site: Optional[SiteVerificationWebResourceGettokenRequestSite] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('site') }})
    verification_method: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('verificationMethod') }})
    

File: SDKs/googleapis.com/siteVerification/v1/typescript/src/sdk/models/shared/siteverificationwebresourcegettokenrequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// SiteVerificationWebResourceGettokenRequestSite
/** 
 * The site for which a verification token will be generated.
**/
export class SiteVerificationWebResourceGettokenRequestSite extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=identifier" })
  identifier?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;
}


export class SiteVerificationWebResourceGettokenRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=site" })
  site?: SiteVerificationWebResourceGettokenRequestSite;

  @SpeakeasyMetadata({ data: "json, name=verificationMethod" })
  verificationMethod?: string;
}
